Source/JavaScriptCore/assembler/ARMv7Assembler.h

```diff
@@ -1372,7 +1372,7 @@ class ARMv7Assembler {
             twoWordOp5i6Imm4Reg4EncodedImmSecond(right, lo16),
             twoWordOp5i6Imm4Reg4EncodedImmFirst(OP_MOVT, hi16),
             twoWordOp5i6Imm4Reg4EncodedImmSecond(right, hi16),
-            static_cast<uint16_t>(OP_CMP_reg_T2 | left)
+            static_cast<uint16_t>(static_cast<uint16_t>(OP_CMP_reg_T2) | static_cast<uint16_t>(left))
```

Comment:
I missed this earlier, but you can remove the outer cast now:

```suggestion
            static_cast<uint16_t>(OP_CMP_reg_T2) | static_cast<uint16_t>(left)
```

Reply:
I tried, but then GCC (12.2.0) warns about a narrowing conversion:

```
Source/JavaScriptCore/assembler/ARMv7Assembler.h:1375:50: warning: narrowing conversion of ‘(int)(((short unsigned int)((int)((uint16_t)JSC::ARMv7Assembler::OP_CMP_reg_T2))) | ((short unsigned int)((int)((uint16_t)left))))’ from ‘int’ to ‘uint16_t’ {aka ‘short unsigned int’} [-Wnarrowing]
 1375 |             static_cast<uint16_t>(OP_CMP_reg_T2) | static_cast<uint16_t>(left)
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```